lab_1/main.py

```diff
@@ -3,81 +3,148 @@
 Language detection
 """
 
+import json
 
-def tokenize(text: str) -> list or None:
+
+def tokenize(text: str):
     """
     Splits a text into tokens, converts the tokens into lowercase,
     removes punctuation and other symbols from words
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
-    """
-    pass
+        """
+    if isinstance(text, str):
+        text = text.lower()
+        string_with_tokens = ""
+        for i in text:
+            if i.isalpha() or i.isspace():
+                string_with_tokens += i
+        tokens = string_with_tokens.split()
+        return tokens
+    return None
 
 
-def remove_stop_words(tokens: list, stop_words: list) -> list or None:
+def remove_stop_words(tokens: list, stop_words: list):
     """
     Removes stop words
     :param tokens: a list of tokens
     :param stop_words: a list of stop words
     :return: a list of tokens without stop words
     """
-    pass
+    try:
+        for i in tokens:
+            if not isinstance(i, str):
+                return None
+            tokens = [i for i in tokens if i not in stop_words]
+            return tokens
+    except TypeError:
+        return None
 
 
-def calculate_frequencies(tokens: list) -> dict or None:
```

Comment:
Why did you remove `-> dict or None`?
these lines are quite useful

change back everywhere